runtime/resolvers/postgres_sql.go

```diff
@@ -0,0 +1,442 @@
+	dbDir := filepath.Join(args.DataDir, strconv.FormatInt(time.Now().UnixMilli(), 10))
+	if err := os.Mkdir(dbDir, fs.ModePerm); err != nil {
+		return nil, err
+	}
+	dbPath := filepath.Join(dbDir, "catalog.db")
+	// nolint:gosec // We don't need cryptographically secure random numbers
+	dbName := fmt.Sprintf("pg_catalog_db_%v", rand.Int())
+	args.SQL = rewriteSQL(args.SQL)
+	// loop over all resources and create corresponding table in duckdb so that these can be queried with information_schema
+	for _, resource := range resources {
+		metricSQL, err := fromQueryForMetricsView(ctx, ctrl, opts, resource)
+		if err != nil {
+			if errors.Is(err, errForbidden) {
+				continue
+			}
+			return nil, err
+		}
+
+		compiler := metricssqlparser.New(ctrl, opts.InstanceID, opts.UserAttributes, args.Priority)
+		createTableSQL, connector, _, err := compiler.Compile(ctx, metricSQL)
+		if err != nil {
+			return nil, err
+		}
+
+		if err := populateMetricView(ctx, opts, connector, createTableSQL, dbPath, dbName, resource.Meta.Name.Name); err != nil {
+			return nil, err
+		}
+	}
+
+	handle, err := drivers.Open("duckdb", opts.InstanceID, map[string]any{"path": dbPath}, activity.NewNoopClient(), zap.NewNop())
+	if err != nil {
+		return nil, err
+	}
+
+	olap, ok := handle.AsOLAP(opts.InstanceID)
+	if !ok {
+		return nil, fmt.Errorf("developer error : handle is not an OLAP driver")
+	}
+	if err := olap.Exec(ctx, &drivers.Statement{Query: "USE catalog.public"}); err != nil {
+		return nil, err
+	}
+
+	if err := populateCatalogTables(ctx, olap); err != nil {
+		return nil, err
+	}
+	fmt.Printf("final SQL: %v\n", args.SQL)
+	return &catalogSQLResolver{
+		olap:  olap,
+		sql:   args.SQL,
+		dbDir: dbDir,
+	}, nil
+}
+
+func populateMetricView(ctx context.Context, opts *runtime.ResolverOptions, connector, query, path, dbName, mvName string) error {
+	olap, release, err := opts.Runtime.OLAP(ctx, opts.InstanceID, connector)
+	if err != nil {
+		return err
+	}
+	defer release()
+
+	if olap.Dialect() != drivers.DialectDuckDB {
+		// other OLAP engines can be supported with a schema translation layer from engine -> duckDB
+		// and creating table with DDL statements
+		return fmt.Errorf("only duckdb is supported")
+	}
+
+	err = olap.WithConnection(ctx, 1, false, false, func(ctx, ensuredCtx context.Context, _ *sql.Conn) error {
+		if err := olap.Exec(ctx, &drivers.Statement{Query: fmt.Sprintf("ATTACH '%s' AS %s", path, dbName)}); err != nil {
+			return err
+		}
+
+		// postgres's default schema is public
+		if err := olap.Exec(ctx, &drivers.Statement{Query: fmt.Sprintf("CREATE SCHEMA IF NOT EXISTS %s.public", dbName)}); err != nil {
+			return err
+		}
+
+		defer func() {
+			_ = olap.Exec(ensuredCtx, &drivers.Statement{Query: fmt.Sprintf("DETACH %s", dbName)})
+		}()
+
+		qry := fmt.Sprintf("CREATE TABLE %s.public.%s AS SELECT * FROM (%s) LIMIT 0", dbName, olap.Dialect().EscapeIdentifier(mvName), query)
+		return olap.Exec(ctx, &drivers.Statement{Query: qry})
+	})
```

Comment:
Since this was a MVP the idea was to avoid schema translation for each of the OLAP drivers and use duckdb's internal logic to create a table with 0 rows which can be queried. The easiest way was to `ATTACH` the DB to main DB and fire `CREATE TABLE AS SELECT ...` query.
I can remove this logic if we want to make it work for all OLAP engines right now:

1. I didn't understand this point. Do you mean we cache the schema of the metric view in the state? I think in all cases we would still need to generate the `SELECT dim, measures ....` SQL to handle security policies based on user attributes?

2. This sounds like an easy solution that should work for all types if we are okay with the approximation. The issue is that if user tries to cast to fix this, it will again fail.

Reply:
I think this is worth fixing if only because `ATTACH`ing from multiple DBs feels like a way we might end potentially up with another WAL explosion issue or similar issue. 

But it's also likely that the first users of the PG wire compatibility will be users who are not on DuckDB.

For 1, yes, but then you could just iterate over the types in the state and quickly generate the `SELECT ...` here (applying the security policy to exclude fields of course).

But 2 also seems good for now. As you said, it's just for an MVP.